src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,95 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    public static final int DEFAULT_INITIAL_CAPACITY = 16;
+    public static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    public static final int CAPACITY_INCREASER = 2;
+
+    Node<K, V>[] bucketArray;
+    private int size;
+
+    public MyHashMap() {
+        bucketArray = new Node[DEFAULT_INITIAL_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
+        if (size + 1 > bucketArray.length * DEFAULT_LOAD_FACTOR) {
+            resize();
+        }
 
+        int bucket = bucket(key);
+        if (bucketArray[bucket] == null) { //empty bucket
+            bucketArray[bucket] = new Node<>(key, value);
+            size++;
+            return;
+        }
+
+        Node<K, V> current = bucketArray[bucket];
+        for (int i = 0; i < size; i++) {
+            if (key == null ? key == current.key : key.equals(current.key)) {
```

Comment:
In this, if case, u used the same logic. if key null, you check is key == current.key, and if key not null, you do the same thing. 
```suggestion
            if (Objects.equals(key, current.key)) {
```